### PR TITLE
vdirsyncer: Convert to a python module

### PR DESCRIPTION
**Pull request by doronbehar**
###### Motivation for this change

`khal`, is importing `vdirsyncer` as a python module in it's checkPhase. However, as of currently, khal is being called with the argument `pkgs`and then `khal` is using `pkgs.vdirsyncer` in it's `checkInputs`. This is problematic if you want to write an overlay that would (e.g):

- Use python38 for khal.
- vdirsyncer is overridden to use Python38.

Without this patch, doing either of the following will make the build of khal fail. It was partially discussed in https://github.com/NixOS/nixpkgs/pull/85993 but this PR is smaller so hopefully it'll be easier to review it.

Plus that, `vdirsyncerStable` seems pretty broken, and it was marked as so accordingly.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by FRidh** [comment]
We typically avoid having multiple versions, especially in `python-packages.nix`. If the stable one is broken already, maybe remove it?

**Review by FRidh — APPROVED** [review]

**Comment by doronbehar** [comment]
> If the stable one is broken already, maybe remove it?

Sounds good to me. Shall we rename it as well then? Currently the version we actually use for the unstable, non broken vdirsyncer is here: https://github.com/pimutils/vdirsyncer/compare/master...spk:fix-build-style

I also saw @marsam's commits appear there so maybe they have an opinion.

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -1607,6 +1607,13 @@ in {
 
   vdf = callPackage ../development/python-modules/vdf { };
 
+  vdirsyncer = disabledIf (!isPy3k) (callPackage ../development/python-modules/vdirsyncer {
+    inherit (pkgs.darwin.apple_sdk.frameworks) CoreServices Security;
+    inherit (pkgs) pkg-config openssl rustPlatform;
+  });
+
+  vdirsyncerStable = disabledIf (!isPy3k) (callPackage ../development/python-modules/vdirsyncer/stable.nix { });
```
Comment (jonringer) [inline pkgs/top-level/python-packages.nix:1615]: please move disable logic inside the expression
```suggestion
  vdirsyncer = callPackage ../development/python-modules/vdirsyncer {
    inherit (pkgs.darwin.apple_sdk.frameworks) CoreServices Security;
    inherit (pkgs) pkg-config openssl rustPlatform;
  };

  vdirsyncerStable = callPackage ../development/python-modules/vdirsyncer/stable.nix { };
```

File: pkgs/development/python-modules/vdirsyncer/stable.nix [hunk 2]
```diff
@@ -1,27 +1,53 @@
...
+, pytest-subtesthack
+, setuptools_scm
+}:
 
 # Packaging documentation at:
 # https://github.com/pimutils/vdirsyncer/blob/0.16.7/docs/packaging.rst
-python3Packages.buildPythonApplication rec {
+buildPythonPackage rec {
   version = "0.16.7";
   pname = "vdirsyncer";
```
Comment (jonringer) [inline pkgs/development/python-modules/vdirsyncer/stable.nix:26]:
```suggestion
  pname = "vdirsyncer";
  disabled = isPy27;
```

File: pkgs/development/python-modules/vdirsyncer/stable.nix [hunk 3]
```diff
@@ -1,27 +1,53 @@
-{ lib, python3Packages, fetchpatch }:
+{ stdenv
+, pythonAtLeast
+, buildPythonPackage
+, fetchPypi
```
Comment (jonringer) [inline pkgs/development/python-modules/vdirsyncer/stable.nix:4]:
```suggestion
, fetchPypi
, isPy27
```

File: pkgs/development/python-modules/vdirsyncer/default.nix [hunk 4]
```diff
@@ -1,8 +1,31 @@
...
+, setuptools_scm
+}:
 
 # Packaging documentation at:
 # https://github.com/untitaker/vdirsyncer/blob/master/docs/packaging.rst
-python3Packages.buildPythonApplication rec {
+buildPythonPackage rec {
   version = "unstable-2018-08-05";
   pname = "vdirsyncer";
   name = "${pname}-${version}";
```
Comment (jonringer) [inline pkgs/development/python-modules/vdirsyncer/default.nix:31]:
```suggestion
  name = "${pname}-${version}";
  disabled = isPy27;
```

File: pkgs/development/python-modules/vdirsyncer/default.nix [hunk 5]
```diff
@@ -1,8 +1,31 @@
-{ stdenv, python3Packages, fetchFromGitHub, fetchpatch, rustPlatform, pkgconfig, openssl, CoreServices, Security }:
+{ stdenv
+, buildPythonPackage
+, fetchFromGitHub
+, fetchpatch
```
Comment (jonringer) [inline pkgs/development/python-modules/vdirsyncer/default.nix:4]:
```suggestion
, fetchpatch
, isPy27
```

Comment (doronbehar) [reply pkgs/development/python-modules/vdirsyncer/default.nix:5]: :guitar:

File: pkgs/development/python-modules/vdirsyncer/stable.nix (hunk 3 above)
Comment (doronbehar) [reply pkgs/development/python-modules/vdirsyncer/stable.nix:4]: :saxophone:

File: pkgs/development/python-modules/vdirsyncer/stable.nix (hunk 2 above)
Comment (doronbehar) [reply pkgs/development/python-modules/vdirsyncer/stable.nix:27]: :1234:

File: pkgs/top-level/python-packages.nix (hunk 1 above)
Comment (doronbehar) [reply pkgs/top-level/python-packages.nix]: :trumpet:

**Pull request merged**